### PR TITLE
Implement `cfg_attr` handling

### PR DESCRIPTION
**Pull request by jonas-schievink**
Part of https://github.com/rust-analyzer/rust-analyzer/issues/5548

### CODE REVIEW

File: crates/hir_def/src/attr.rs [hunk 1]
```diff
@@ -122,9 +122,67 @@ impl RawAttrs {
...
+                let attr = Subtree { delimiter: None, token_trees: attr };
+                let cfg = Subtree { delimiter: subtree.delimiter, token_trees: cfg };
+                let cfg = CfgExpr::parse(&cfg);
+
+                let cfg_options = &crate_graph[krate].cfg_options;
+                if cfg_options.check(&cfg) == Some(false) {
+                    None
+                } else {
+                    let attr = ast::Attr::parse(&format!("#[{}]", attr)).ok()?;
+                    let hygiene = Hygiene::new_unhygienic(); // FIXME
```
Comment (matklad) [inline crates/hir_def/src/attr.rs:179]: Yeah, this is probably good enough for now :)

File: crates/hir_def/src/nameres/tests/diagnostics.rs [hunk 2]
```diff
@@ -119,3 +119,19 @@ fn inactive_item() {
         "#,
     );
 }
+
+/// Tests that `cfg` attributes behind `cfg_attr` is handled properly.
+#[test]
+fn inactive_via_cfg_attr() {
```
Comment (matklad) [inline crates/hir_def/src/nameres/tests/diagnostics.rs:125]: this could use a `mark`

**Review by matklad — APPROVED** [review]

File: crates/hir_def/src/attr.rs (hunk 1 above)
Comment (jonas-schievink) [reply crates/hir_def/src/attr.rs:182]: Yup, I tried doing this properly, but threading `Hygiene`/`HirFileId` through everything is a *lot* of work

**Pull request merged**